lib/spack/spack/build_systems/perl.py

```diff
@@ -2,73 +2,87 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
-
 import inspect
 import os
 
 from llnl.util.filesystem import filter_file
 
-from spack.directives import extends
-from spack.package_base import PackageBase, run_after
+import spack.builder
+import spack.package_base
+from spack.directives import buildsystem, extends
+from spack.package_base import PackageBase
 from spack.util.executable import Executable
 
+from ._checks import BaseBuilder
+
 
 class PerlPackage(PackageBase):
-    """Specialized class for packages that are built using Perl.
+    """Specialized class for packages that are built using Perl."""
+
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = "PerlPackage"
+    #: Legacy buildsystem attribute used to deserialize and install old specs
+    legacy_buildsystem = "perl"
+
+    buildsystem("perl")
+
+    #: Callback names for build-time test
+    build_time_test_callbacks = ["check"]
 
-    This class provides four phases that can be overridden if required:
+    extends("perl", when="buildsystem=perl")
```

Comment:
I don't think the `Perl` build-system will ever be mixed with others, but if it does this is the correct way of specifying it in the base class. If it does not, this still works since `buildsystem=perl` is implied by `Perl` being the only build-system.

So, since this approach doesn't prevent anything, but leaves open the possibility for further extensions, I used a conditional directive.

Reply:
I assume if you had a package like:

```
class Foo(AutotoolsPackage, PerlPackage):
    ...
```

they would expect that it depends on `perl` by way of inheriting `PerlPackage` (and you'd want to e.g. augment `PERL5LIB`).